pkg/resourcemanager/keyvaults/keyvault.go

```diff
@@ -49,9 +51,12 @@ func getVaultsClient() (keyvault.VaultsClient, error) {
 	return vaultsClient, nil
 }
 
-func getObjectID(ctx context.Context, tenantID string, clientID string) (*string, error) {
+// TODO(creds-refactor): can we just use the tenant and client id from
+// the creds here? Depends whether the client code uses them to
+// override the creds.
+func getObjectID(ctx context.Context, creds config.Credentials, tenantID string, clientID string) (*string, error) {
```

Comment:
This function is used in 2 places. One of those places is `CreateVaultWithAccessPolicies` which as far as I can tell is only used in tests... may be worth moving to tests directly if that's really a test helper method not used at all in production code.

The other location is `ParseAccessPolicy` which seems to use a clientId/tenantId other than the one in the credentials (supplied by the user?)

Reply:
I've removed the comment - I've left the method for now even though it's used from tests. If it was only used in the tests for keyvault I'd move it, but it's kind of tricky to share testing code between two packages.